### DISCUSSION TITLE
osu!catch editor does not have reverse slider creation abilities

### DISCUSSION THREAD

**Discussion by thaboovx** [ppy/osu#26098] [Ideas] [answered]
In osu!stable, reverse sliders create a fruit at the reverse, and prevent the generation of droplets at the location of the fruit before continuing droplets towards the tail:
![image](https://github.com/ppy/osu/assets/154714922/13f7e20d-6c33-4fa5-b218-9270f0623a73)
![image](https://github.com/ppy/osu/assets/154714922/bab6f774-ba8e-4cc9-90d5-7fd8872f29d0)

In osu!lazer, these juicestreams have different properties than juicestreams created in the osu!lazer catch editor. The reverse sliders have their final vertex at the first reverse, but droplets and fruits continue:
![image](https://github.com/ppy/osu/assets/154714922/81838cf4-21ce-4ab3-8354-1c58a513996c)
This is in large contrast to juicestreams created within osu!lazer's editor, as to my knowledge juicestreams and their objects **must** end at the final vertex, and cannot have more than the start and end fruit within the juicestream (video):
https://github.com/ppy/osu/assets/154714922/ccbfb3dd-e258-4888-bf60-1094ff3c735a

It appears that these patterns cannot be created within osu!lazer's editor tools, which feels frustrating. Reverse slider patterns can be modified within osu!lazer's editor and _technically_ can be made by copy and pasting an existing reverse slider pattern from another map. Editing a reverse slider within osu!lazer's editor functionally works fine as it follows expected reverse slider behaviour.

I think it would be beneficial to modify the osu!catch editor's behaviour to allow for this moderately common pattern to be made. I am unsure of how the implementation could work apart from allowing two-vertex juicestreams to have a reverse property and # of reverses.

**Comment by peppy** [comment, marked as answer]
It works the same way as in stable:

https://github.com/ppy/osu/assets/191335/22399ecc-3a7b-48fa-875a-7626fd3f50a1

**Reply by thaboovx** [reply]
Whoops. I'm so used to the osu!standard workflow where reverse sliders are created through playing the song while creating the reverse that I hadn't considered that. Apologies and thanks.
